docs/reference/api/pieces.md

```diff
@@ -0,0 +1,268 @@
+# Piece type REST endpoints
+
+Every [piece type](TODO) has built in REST end points that share their overall structure in common. The exact document properties returned will depend on the piece type's fields.
+
+## Endpoints
+
+**Note:** `:piece-name` represents the `name` option for a piece type. For example, you would request all pieces of an `article` piece type with `/api/v1/article`.
+
+| Method | Path | Description | Auth required |
+|---------|---------|---------|---------|
+|GET | [`/api/v1/:piece-name`](#get-api-v1-piece-name)| Get all pieces of a given type, paginated| FALSE |
+|GET | [`/api/v1/:piece-name/:id`](#get-api-v1-piece-name-id)| Get a single piece with a specified ID | FALSE |
+|POST | [`/api/v1/:piece-name`](#post-api-v1-piece-name)| Insert a new piece of the specified type | TRUE |
+|PUT | [`/api/v1/:piece-name/:id`](#put-api-v1-piece-name-id)| Fully replace a specific piece document | TRUE |
+|PATCH | [`/api/v1/:piece-name/:id`](#patch-api-v1-piece-name-id)| Update only certain fields on a specific document | TRUE |
+|DELETE | Not supported | Instead `PATCH` the `trash` property to `true` | n/a |
+
+**This guide will use an `article` piece type as an example.** In addition to standard piece fields, this hypothetical piece type has the following fields (for the sake of illustration):
+- `author`: a `relationship` field connected to the `user` piece type
+- `category`: a String field
+- `body`: an `area` field using the rich text widget.
+
+## `GET: /api/v1/:piece-name`
```

Comment:
`piece-module-name` might be best.

Reply:
Why is this better, out of curiosity.

Reply:
Because it is literally the name of the module, and we don't spend any time separately introducing the "name" option anymore because starting with 3.x it is equal to the name of the module (and thank the stars for that).

Reply:
We're generally trying avoid mentioning how everything is a module too often. But I do see how `piece-name` could suggest a piece document's name for folks in-the-know. I could change this to `:piece-type`.